### DISCUSSION TITLE
Question: How to use Alignment only?

### DISCUSSION THREAD

**Discussion by thomasf1** [m-bain/whisperX#1250] [Q&A]
We are trying to use whisperX to align text. We already know the script and just want the start/end.

It should be easy, but the whisperx.align expects the start/end as well as the text as an input.

```python
result = whisperx.align([{'text': 'My text..', 'start': 0.268, 'end': 10.86}], model_a, metadata, audio, device, return_char_alignments=False)
```

Here is what we tried:
- Detection via whisper, leaving the start/end and replacing text -> works but wasteful/slow as only start/end is used
- Using whisperx.align with a start of 0 and end of the audio length -> kinda works, but the confidence scores drop quite a bit - the start value is changed, but different from the detected one
- Trying to extract the start/end from faster_whisper without actual detection, but to be honest, we got a bit stuck there:

```python
sampling_rate = self.feature_extractor.sampling_rate

if not isinstance(audio, np.ndarray):
	audio = decode_audio(audio, sampling_rate=sampling_rate)

duration = audio.shape[0] / sampling_rate

vad_parameters = VadOptions()
speech_chunks = get_speech_timestamps(audio, vad_parameters)
```
as somehow we got other start/end times. Not quite sure what´s the easiest way to do it, kinda having the feeling we miss something obvious...

PS: Thanks for the great project, using this helped us a lot. :)
PS: Also, an option to turn off the vad filter might be nice (we have well structured data that is already broken down into small chunks of text).

**Comment by m-bain** [comment]
Hi, the align function takes start and end times because it is not recommended to align text with long audio with the current code.

1. DTW alignment on long sequences is prone to drifting and expect a drop in accuracy

2. Memory constraints, wav2vec2 with transformer architecture leads to large memory for long input audio.

> Using whisperx.align with a start of 0 and end of the audio length -> kinda works, but the confidence scores drop quite a bit - the start value is changed, but different from the detected one

You can try denoising the audio using demucs (wav2vec2 is better on clean audio) e.g. https://github.com/facebookresearch/demucs

If your audio is short, then you can give start time as 0 and end time as full audio length

> PS: Also, an option to turn off the vad filter might be nice (we have well structured data that is already broken down into small chunks of text).

The VAD is to ensure the audio is not sliced in the middle of words, since this would cause boundary errors.

**Comment by lokesh1199** [comment]
What will be a ideal length for the input audio to get accurate word level alignment using whisperX?

**Comment by mbdwwlc2020** [comment]
Can I use whisperx for alignment only? For example, I have *.mp4 files and *.srt files, but srt and mp4 are not synchronized. Can I use whisperx to automatically modify the srt timesline so that it can be aligned to mp4?  thx

**Comment by Manamama** [comment]
> Can I use whisperx for alignment only? For example, I have *.mp4 files and *.srt files, but srt and mp4 are not synchronized. Can I use whisperx to automatically modify the srt timesline so that it can be aligned to mp4? thx

I guess one should rerun whisperx on whole audio and have some LLM match the new SRT (with the "right" timecodes) against the older SRT (where the timecodes were off but the text was presumably better quality, e.g. manually fixed), as a kind of 'meld' operation . And then have the LLM marry the two.
